Odata-docs/webapi-8/fundamentals/actions-functions.md

```diff
@@ -0,0 +1,409 @@
+            "title": "Book 2",
+            "year": 2001,
+            "forKids": true
+        },
+        {
+            "id": "4",
+            "isbn": "DD0011",
+            "title": "Book 4",
+            "year": 2003,
+            "forKids": true
+        },
+        {
+            "id": "5",
+            "isbn": "EE0011",
+            "title": "Book 5",
+            "year": 2004,
+            "forKids": true
+        },
+        {
+            "id": "6",
+            "isbn": "FF0011",
+            "title": "Book 6",
+            "year": 2005,
+            "forKids": true
+        },
+        {
+            "id": "7",
+            "isbn": "GG0011",
+            "title": "Book 7",
+            "year": 2006,
+            "forKids": true
+        }
+    ]
+}
+```
+
+## Bound Action
+Similar to bound functions, bound actions can be bound to entity type or collection of entity type.
+However, the overload of bound actions are different. Bound actions can be overloaded, but overload must happen by different binding parameter. For one binding parameter, there can be only one bound action.
+
+In the Edm Model, we modify the code as follows.
+
+```csharp
+private static IEdmModel GetEdmModel()
+{
+    var builder = new ODataConventionModelBuilder();
+    builder.EntitySet<Book>("books");
+    builder.EntityType<Book>().Collection
+        .Function("mostRecent")
+        .Returns<string>();
+    builder.Function("returnAllForKidsBooks").ReturnsFromEntitySet<Book>("books");
+    builder.EntityType<Book>()
+        .Action("rate")
+        .Parameter<int>("rating");
+    var model = builder.GetEdmModel();
+    return model;
+}
+```
+
+In the Edm Model above, we are adding a `rate` action which is bound to a `Book` entity type. The action takes a `rating` parameter of type `int`. 
+
+We will update the `BooksController` as follows.
+
+```csharp
+public class BooksController : ODataController
+{
+    // Get ~/Books
+    [EnableQuery]
+    public IActionResult Get()
+    {
+        return Ok(DataSource.Instance.Books);
+    }
+
+    [HttpGet("odata/Books/MostRecent()")]
+    public IActionResult MostRecent()
+    {
+        var maxBookId = DataSource.Instance.Books.Max(x => x.ID);
+        return Ok(maxBookId);
+    }
+
+    [HttpGet("odata/ReturnAllForKidsBooks")]
+    public IActionResult ReturnAllForKidsBooks()
+    {
+        var forKidsBooks = DataSource.Instance.Books.Where(m => m.ForKids == true);
+        return Ok(forKidsBooks);
+    }
+
+    [HttpPost("odata/Books({key})/Rate")]
+    public IActionResult Rate([FromODataUri] string key, ODataActionParameters parameters)
+    {
+        if (!ModelState.IsValid)
+        {
+            return BadRequest();
+        }
+
+        int rating = (int)parameters["rating"];
+
+        if (rating < 0)
+        {
+            return BadRequest();
+        }
+
+        return Ok(new BookRating() { BookID = key, Rating = rating });
+    }
+}
+```
+
+If we invoke `POST odata/Books('1')/Rate {"rating": 7}`
```

Comment:
I think we should heave the request URL and body on separate line to make it clear the the JSON part is the body:

```
POST odata/Books('1')/Rate

{"rating": 7}
```